code/modules/mob/mob_movement.dm

```diff
@@ -604,3 +604,37 @@
 			. = T.adjust_slowdown(src, .)
 		if(movement_speed_modifier)
 			. *= (1/movement_speed_modifier)
+
+
+
+
+/mob/living/carbon/proc/toggle_move_intent()
+	if(src.legcuffed)
+		to_chat(src, "<span class='notice'>You are legcuffed! You cannot run until you get [src.legcuffed] removed!</span>")
+		src.m_intent = M_INTENT_WALK	//Just incase
+		src.hud_used.move_intent.icon_state = "walking"
+		return 1
+	switch(usr.m_intent)
```

Comment:
remove all these `usr.`, so that this becomes `switch(m_intent)`, for instance.
do this to all lines inside this proc